sos/plugins/etcd.py

```diff
@@ -23,15 +23,46 @@ class etcd(Plugin, RedHatPlugin):
     """etcd plugin
     """
 
+    packages = ('etcd',)
+
     def setup(self):
-        self.add_copy_spec("/etc/etcd")
+        self.address = self.get_address()
+        self.cmd = 'etcdctl '
+
+        self.add_copy_spec('/etc/etcd')
+
+        subcmds = ['--version',
+                   'member list',
+                   'cluster-health',
+                   'ls --recursive',
+                   ]
+
+        for sub in subcmds:
+            self.add_cmd_output("%s %s" % (self.cmd, sub))
+
+        api_urls = ['/v2/stats/leader',
+                    '/v2/stats/self',
+                    '/v2/stats/store',
+                    ]
+
+        for url in api_urls:
+            self.add_cmd_output("curl -s %s%s" % (self.address, url))
 
-        self.add_cmd_output("curl http://localhost:4001/version")
-        self.add_cmd_output("curl http://localhost:4001/v2/members")
-        self.add_cmd_output("curl http://localhost:4001/v2/stats/leader")
-        self.add_cmd_output("curl http://localhost:4001/v2/stats/self")
-        self.add_cmd_output("curl http://localhost:4001/v2/stats/store")
         self.add_cmd_output("ls -lR /var/lib/etcd/")
 
+    def get_address(self):
+        try:
+            with open('/etc/etcd/etcd.conf', 'r') as ef:
+                for line in ef:
+                    if line.startswith('ETCD_LISTEN_CLIENT_URLS'):
+                        return line.split('=')[1].replace('"', '').strip()
+        # If we can't read etcd.conf, assume defaults by etcd version
+        except IOError:
```

Comment:
I can't really think of any other we would hit here, so I made it a generic `except` catch.